Lib/asyncio/pools.py

```diff
@@ -0,0 +1,166 @@
+"""Support for high-level asynchronous pools in asyncio."""
+
+__all__ = 'ThreadPool',
+
+
+import concurrent.futures
+import functools
+import threading
+import os
+
+from . import events
+from . import exceptions
+from . import futures
+
+
+class AbstractPool:
+    """Abstract base class for asynchronous pools."""
+
+    async def astart(self):
+        raise NotImplementedError
```

Comment:
Is it possible to decorate this method with `abc.abstractmethod`, so that the base class can't be instantiated?

Reply:
Yeah, that seems like a good idea and would probably be more explicit. The only reason that I didn't use it was just that I modeled it to be similar to the existing abstract classes in asyncio, such as `AbstractServer` and `AbstractEventLoopPolicy`.

But, I see no reason not to add `abc.abstractmethod`. I can update it accordingly.

Reply:
Done.